packages/cozy-client/src/hooks/useQuery.js

```diff
@@ -7,8 +7,10 @@ const resolveToValue = fnOrValue => {
   return typeof resolveToValue === 'function' ? fnOrValue() : fnOrValue
 }
 
-const generateFetchMoreQueryDefinition = collection => {
-  return collection.definition.offsetBookmark(collection.bookmark)
+const generateFetchMoreQueryDefinition = query => {
+  return query.bookmark
+    ? query.definition.offsetBookmark(query.bookmark)
+    : query.definition.offset(query.data.length)
```

Comment:
s/query/queryResult would make it a bit more clear, the naming is always a bit difficult between query result and queries.